sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice.json

```diff
@@ -1,3093 +1,3258 @@
+                        "failed",
+                        "cancelled",
+                        "scheduled",
+                        "queued"
+                    ],
+                    "type": "string"
+                },
+                "failureReason": {
+                    "description": "System generated.  Ignored at creation.\r\nIf status == failure, this represents a string containing the reason.",
+                    "type": "string"
+                },
+                "statusMessage": {
+                    "description": "Status message for the job",
+                    "type": "string"
+                },
+                "deviceJobStatistics": {
+                    "$ref": "#/definitions/DeviceJobStatistics",
+                    "description": "Job details"
+                }
+            }
+        },
+        "DeviceJobStatistics": {
+            "description": "The job counts, e.g., number of failed/succeeded devices",
+            "type": "object",
+            "properties": {
+                "deviceCount": {
+                    "format": "int32",
+                    "description": "Number of devices in the job",
+                    "type": "integer"
+                },
+                "failedCount": {
+                    "format": "int32",
+                    "description": "The number of failed jobs",
+                    "type": "integer"
+                },
+                "succeededCount": {
+                    "format": "int32",
+                    "description": "The number of Successed jobs",
+                    "type": "integer"
+                },
+                "runningCount": {
+                    "format": "int32",
+                    "description": "The number of running jobs",
+                    "type": "integer"
+                },
+                "pendingCount": {
+                    "format": "int32",
+                    "description": "The number of pending (scheduled) jobs",
+                    "type": "integer"
+                }
+            }
+        },
+        "QueryResult": {
+            "description": "The query result.",
+            "type": "object",
+            "properties": {
+                "type": {
+                    "description": "The query result type.",
+                    "enum": [
+                        "unknown",
+                        "twin",
+                        "deviceJob",
+                        "jobResponse",
+                        "raw",
+                        "enrollment",
+                        "enrollmentGroup",
+                        "deviceRegistration"
+                    ],
+                    "type": "string"
+                },
+                "items": {
+                    "description": "The query result items, as a collection.",
+                    "type": "array",
+                    "items": {
+                        "type": "object"
+                    }
+                },
+                "con
```

Comment:
We will be hiding this property from the user. The `destination` needs to be in the format: `/devices/{deviceId}/messages/deviceBound`, and can be constructed internally by the SDK, since we know the deviceId.

Reply:
Shouldn't this be baked into the path of the API itself instead of this object definition? Or does the service ask for it in the payload somewhere?

Reply:
They require it as a part of the http request. This is because, their path currently does not contain any device info, it is simply `/messages/devicebound`.